composer/datasets/in_context_learning_evaluation.py

```diff
@@ -82,826 +109,900 @@ def _make_padded_input(context_enc, continuation_enc, max_seq_len, pad_tok_id, p
-            inp, _ = _make_padded_input(context_enc, [],
-                                        self.max_seq_len - self.max_answer_length,
-                                        self.pad_tok_id,
-                                        padding_side=self.padding_side)
+        Args:
+            example (Dict): A Dictionary from the hf dataset
+            example_idx (int): the index of example
+            num_fewshot (int): Number of examples context/continuation pairs to prepend to the test pair
+            prompt_string (str): The prompt to prepend to all inputs
+            fewshot_rng (random.Random): Random number generator to use for fewshot sampling
 
-            inputs.append(inp)
-            answers.append(aliases)
+        Returns:
+            Dict: contains a dictionary with the tokenized data
+        """
+        prompt_and_fewshot = self._generate_few_shot_prompt(num_fewshot, example_idx, prompt_string, fewshot_rng)
+        ctxt = self._construct_context(example, prompt_and_fewshot, add_answer=False)
+        tokenized_example = self._tokenize_example(prompt_and_fewshot, ctxt, example)
+        return tokenized_example
+
+    # TODO: Maybe make this not a class function. Also, could make our padding operations work on lists
+    def _convert_tokens_to_tensors(self, batch: Dict) -> Dict[str, Any]:
+        # zzzz HF converts ur torch tensors into lists so need to convert them back
+        batch['input_ids'] = torch.stack(list(map(torch.tensor, batch['input_ids'])))
+        if self.tokenize_labels:
+            batch['labels'] = torch.stack(list(map(torch.tensor, batch['labels'])))
+            batch['continuation_indices'] = list(map(torch.tensor, batch['continuation_indices']))
+        return batch
 
-            # We will search for the answer within the portion of the model response
-            # beginning with `cot_delimiter`
-            cot_delimiter = sample['cot_delimiter']
+    def collate_fn(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        The function that the dataloader uses to accumulate data into batches.
+        Args:
+            data (List): list of tokenized datapoints (dicts returned by self._tokenize_example)
 
-        batch = {
-            'input_ids': torch.stack(inputs),
-            'mode': 'generate',
-            'labels': answers,
-            'cot_delimiter': cot_delimiter,
-            'generation_length': self.max_answer_length,
-            'generation_kwargs': {
-                'pad_token_id': self.pad_tok_id,
-                'use_cache': True
-            }
-        }
+        Returns:
+            Dict: dictionary for a single batch
+        """
+        batch = copy.deepcopy(self.base_batch)
+        for data_pair in data:
+            for batch_key, data_key in self.batch_mapping.items():
+                batch[batch_key].append(data_pair[data_key])
+            if 'continuation_indices' in data_pair:
```

Comment:
why can't this be part of the loop on 498-499?

Reply:
"continuation_indices" is left out of the batch_mapping because it's automatically added if tokenize_labels is true, and I don't want the user to have to think about it, so I don't want it to be part of self.batch_mapping

Reply:
Ah yes I forgot about QA and codegen tasks